### PR TITLE
count free space implementation

### LINKED ISSUE #13: 8. Task: Player (availableSpace)
W klasie Player napisz prywatną funkcję, która obliczać będzie availableSpace_ na podstawie aktualnej ilości towaru na statku.

termin - 17.06.2020

### PR DESCRIPTION
**Pull request by Marcinati**
I'm not sure if we need to go deeper in getAllCargo container?

### CODE REVIEW

File: shm/player.cpp [hunk 1]
```diff
@@ -21,3 +22,8 @@ Cargo Player::getCargo(size_t index) {
 
     return 0;
 }
+
+size_t countFreeSpace() { 
+    size_t loadedSpace = std::accumulate(ship->getAllCargo.begin(), ship->getAllCargo.end(), 0);
+    return ship->getCapacity() - loadedSpace;
+}
```
Comment (mateusz-talma) [inline shm/player.cpp:32]: Fine solution with std::accumulate :) Please change "ship" to "ship_". Should we check if ship is nullptr?

**Review by mateusz-talma — DISMISSED** [review]
Please read my comment for countFreeSpace()

**Comment by mateusz-talma** [comment]
> I'm not sure if we need to go deeper in getAllCargo container?


I think it's a good solution

File: shm/player.cpp (hunk 1 above)
Comment (Marcinati) [reply shm/player.cpp:32]: good point

**Review by WBigaj — APPROVED** [review]

**Review by NeoBauer — APPROVED** [review]

**Pull request merged**